src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -625,7 +648,7 @@ else if (!allowFiltering)
                     if (vectorColumn.isPresent())
                     {
                         var vc = vectorColumn.get();
-                        var hasIndex = indexRegistry.listIndexes().stream().anyMatch(i -> i.dependsOn(vc));
+                        var hasIndex = indexRegistry.listIndexes().stream().anyMatch(i -> i.dependsOn(vc)); // TODO CNDB-13129: check excluded hints
```

Comment:
Is this TODO still relevant?

Reply:
It is! I had forgotten about it. Considering hints here can be used to skip vector index column validation if the index is excluded. Creating an index on a vector column shades equality queries with `ALLOW FILTERING` that would normally work without the index. Index hints should be able to disambiguate this:
```
CREATE TABLE t (k int PRIMARY KEY, v vector<float, 2>);
SELECT k FROM t WHERE v = [0.1, 0.2] ALLOW FILTERING; -- works
CREATE CUSTOM INDEX idx ON t(v) USING 'StorageAttachedIndex';
SELECT k FROM t WHERE v = [0.1, 0.2] ALLOW FILTERING; -- doesn't work anymore!
SELECT k FROM t WHERE v = [0.1, 0.2] ALLOW FILTERING  WITH excluded_indexes={idx}; -- works again
```
I've just pushed a fix.